sklearn/preprocessing/_discretization.py

```diff
@@ -188,6 +188,12 @@ def fit(self, X, y=None):
         self.bin_edges_ = bin_edges
         self.n_bins_ = n_bins
 
+        if self.encode != 'ordinal':
+            encode_sparse = self.encode == 'onehot'
+            self.ohe_encoder_ = OneHotEncoder(
```

Comment:
I don't think that we want this attribute to be public.
So it should be named `_ohe_encoder_`